### ISSUE TITLE
Latest release 4.22.0 very unstable - docker desktop mac

### ISSUE THREAD

**Issue by paro-paro** [docker/for-mac#6953]
### Description

Just updated to the latest 4.22.0 version.

Half of my images wont run. Like this one:

```bash
docker run --detach --publish 8545:8545 trufflesuite/ganache:latest
```

Both the cli and the UI have frozen many times trying to run and remove both containers and images. 

Even after a full cleanup with `docker system prune` the UI will still display unused images. 

Just wanted to report this here in case other users are having the same issues. 

Uninstalling and going back to previous version (4.21.1) fixed all issues.

### Reproduce

Explained above

### docker version

```bash
Already rollbacked to the 4.21.1 version where everything is working as expected
```

### docker info

```bash
Same
```

### Diagnostics ID

-

**Comment by brokengrappler** [comment]
I'm experiencing the same. After I updated and installed nothing else, ```docker ps``` doesn't even work for me. It just hangs.

**Comment by bsousaa** [comment]
We released the 4.22.1 patch fixing multiple issues https://docs.docker.com/desktop/release-notes/. Can you give it a try?

**Comment by mccolljr** [comment]
I am experiencing the same problem(s) described by @brokengrappler and @paro-paro on 4.22.1.

My system:

```
macOS version:

13.3.1 (a) (22E772610a)

hardware:

Model Name: MacBook Pro
Model Identifier: Mac14,10
Model Number: MNW83LL/A
Chip: Apple M2 Pro
Total Number of Cores: 12 (8 performance and 4 efficiency)
Memory: 16 GB
System Firmware Version: 8422.100.650
OS Loader Version: 8422.100.650
```

**Comment by daveisfera** [comment]
F14EC0BF-E65E-45F4-B960-78F5BD1C8BD8/20230828180056

**Comment by bsousaa** [comment]
@daveisfera can you confirm what version of Docker Desktop you are running and describe the issue?

**Comment by daveisfera** [comment]
4.22.0 and everything froze up. Even `docker ps` would just hang and not return anything

**Comment by stevehs17** [comment]
I am running 4.22.0 on a mac. It seems flaky. For instance, running a tutorial, sometimes `docker build -t getting-started` works immediately and sometimes it hangs or takes a long time.

**Comment by bsousaa** [comment]
@stevehs17 please upgrade to 4.22.1. @daveisfera can you confirm if the issues are gone?

**Comment by daveisfera** [comment]
The issue is infrequent and random, and I've only been running for about a day, so can't say anything sure yet, but I'll keep an eye out

**Comment by stevehs17** [comment]
@bsousaa I had similar problems yesterday with 4.22.1.

**Comment by savamilovanovic** [comment]
I upgraded to 4.22.1 and it is very unstable - at least as unstable as it was 4.22.0

**Comment by stevehs17** [comment]
After having problems with both 4.22.0 and 4.22.1 on Mac, I decided to try 4.21.1 based on comments previously in this thread. It seems to work much better. One clear advantage is that it starts much more quickly.

**Comment by brokengrappler** [comment]
I went back to 4.21.0 as well. All is running fine as it did before without issues.

**Comment by savamilovanovic** [comment]
I rolled back to 4.22.1. works like a champ!

**Comment by daveisfera** [comment]
docker crashed in a way I haven't see before:
```
request returned Internal Server Error for API route and version http://%2FUsers%2Fdlj%2F.docker%2Frun%2Fdocker.sock/v1.24/containers/json, check if the server supports the requested API version
```

F14EC0BF-E65E-45F4-B960-78F5BD1C8BD8/20230831205555

**Comment by jpmc** [comment]
I don't have a specific log, however I just wanted to add similar confirmation like Steve's.

> After having problems with both 4.22.0 and 4.22.1 on Mac, I decided to try 4.21.1 based on comments previously in this thread. It seems to work much better. One clear advantage is that it starts much more quickly.

Same, and my problems were things like not being able to use docker-ps/cli etc, and my tools that work with docker couldn't detect the instance despite Desktop running, and restarting Docker Desktop in-app didn't make them work either. Only a full system reboot got it to work for a bit before it went weird again eventually.

My only guess at the moment is perhaps it's that new power saving mode? But that's a wild guess on my end. In any case 4.21.1 works fine for me. On an M1 Mac running Monterey.

**Comment by daveisfera** [comment]
F14EC0BF-E65E-45F4-B960-78F5BD1C8BD8/20230920193202
Still having issues with 4.23

**Comment by apanasenkamaksim** [comment]
ERROR: request returned Internal Server Error for API route ...
Docker version 24.0.6, build ed223bc
ProductName:		macOS
ProductVersion:		13.2.1
BuildVersion:		22D68

**Comment by bsousaa** [comment]
@manvydas-tbol @apanasenkamaksim can you share a diagnostics id? https://docs.docker.com/desktop/troubleshoot/overview/

**Comment by apanasenkamaksim** [comment]
@bsousaa Error uploading: Put "https://docker-pinata-support.s3.amazonaws.com/incoming/3/649EA5DE-AE01-4F94-8CA2-DD3BF2EAEA0E/20230922172459.zip": write unix ->/Users/myusername/Library/Containers/com.docker.docker/Data/httpproxy.sock: write: no buffer space available

Maybe this is the issues itself.
Trying to understand how to solve this.

**Comment by apanasenkamaksim** [comment]
Maybe, not.
Diagnostics ID:     649EA5DE-AE01-4F94-8CA2-DD3BF2EAEA0E/20230922173757 (uploaded)

**Comment by daveisfera** [comment]
Froze up again and all 8 cores going full bore in the virtualization stuff:
F14EC0BF-E65E-45F4-B960-78F5BD1C8BD8/20230926210132

**Comment by lorenrh** [comment]
Hello,

This issue has been fixed in the latest 4.24.0 release. You can see the latest release notes [here](https://docs.docker.com/desktop/release-notes/). Would you mind trying it out and checking if the issue still persists?

**Comment by apanasenkamaksim** [comment]
Hi, Thank you!
I updated Docker Desktop and will let you know if the issue appear again.
Thanks again!
